File: src/templates/scripts/receiver.py
#!/usr/bin/env python3
import rospy
from std_msgs.msg import String

class Receiver:

    def __init__(self):
        self.sub = rospy.Subscriber(name='/example_topic', data_class=String, callback=self.callback, queue_size=10)

    def callback(self, msg):
        rospy.loginfo(msg=f'Receiving {msg.data}')
        

if __name__ == '__main__':
    rospy.init_node(name='receiver')
    Receiver()
    rospy.spin()

File: src/templates/src/receiver.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

class Receiver
{
    public:

    Receiver()
    {
        sub = nh.subscribe("/example_topic", 10, &Receiver::callback, this);
    }

    private:

    void callback(const std_msgs::String::ConstPtr &msg)
    {
        ROS_INFO("Receiving %s", msg->data.c_str());
    }

    ros::NodeHandle nh;
    ros::Subscriber sub;

};

int main(int argc, char **argv)
{
    ros::init(argc, argv, "receiver");
    Receiver receiver;
    ros::spin();
    
    return 0;
}